src/region.js

```diff
@@ -69,25 +87,33 @@ function Region({ region }) {
   }, [region]);
 
   useEffect(() => {
-    const findCurrent = slides.find((slide) => currentExecutionId === slide.executionId);
+    const findCurrent = slides.find((slide) => currentSlide?.executionId === slide.executionId);
 
     if (!findCurrent && slides?.length > 0) {
-      const nextSlide = slides[0];
-      setCurrentExecutionId(nextSlide.executionId);
+      const slide = slides[0];
+      setCurrentSlide(slide);
+    }
+
+    const findNext = slides.find((slide) => nextSlide?.executionId === slide.executionId);
```

Comment:
Is this findNext block different from the function "findNextSlide"?
Can it use the same function?